pkg/loadbalancer/loadbalancer.go

```diff
@@ -789,6 +815,13 @@ func (a *L3n4Addr) IsIPv6() bool {
 	return a.AddrCluster.Is6()
 }
 
+// ProtoEqual returns true if protocols match for both L3 and L4.
+func (l *L3n4Addr) ProtoEqual(o *L3n4Addr) bool {
```

Comment:
Since this tests both L3 and L4, perhaps the name `ProtocolsEqual` (note the plural Protocol**s**) would be more self-explanatory. Also, "Proto" is a common abbreviation for "Protocol Buffer", so personally I'd not avoid abbreviating here.